lib/cloudant.js

```diff
@@ -745,7 +745,9 @@ Cloudant.prototype.updateOrCreate = function(model, data, cb) {
   var id = data[idName];
   if (id) {
     self.getCurrentRevision(model, id, function(err, rev) {
-      if (err) return cb(err);
+      if (err && err.statusCode !== 404) {
+        return cb(err);
+      }
```

Comment:
What happens if I do this and I end up getting 200 OK?
The record exists, and now we're calling create on top of it.

Should the body of this function not be something like this?
```
self.updateAttributes(model, id, data, options, function(err, result) {
  if (err) {
     if (err.statusCode === 404) {
        // create logic here
     } else {
        return cb(err);
     }
  } else {
     // do whatever else you need to do, then return results.
  }
}
```